src/debugging/Inspector/scripts/generate.py

```diff
@@ -0,0 +1,197 @@
+        return match.group(1).upper()
+
+    # Split on hyphen, introduce camelcase, and force uppercasing of acronyms.
+    subwords = map(ucfirst, enum_value.split('-'))
+    return re.sub(re.compile(regex, re.IGNORECASE), replaceCallback, "".join(subwords))
+
+def ts_name_for_primitive_type(domain, parameter_type):
+    type_raw_name = ''
+
+    if isinstance(parameter_type, ObjectType) or isinstance(parameter_type, AliasedType):
+        if parameter_type.type_domain() == domain:
+            type_raw_name = parameter_type.raw_name()
+        else:
+            parameter_args = {
+                'domain': parameter_type.type_domain().domain_name,
+                'type': parameter_type.raw_name()
+            }
+            type_raw_name = '%(domain)sDomain.%(type)s' % parameter_args
+    elif isinstance(parameter_type, ArrayType):
+        type_raw_name = '%s[]' % ts_name_for_primitive_type(domain, parameter_type.element_type)
+    elif isinstance(parameter_type, EnumType):
+        if parameter_type.is_anonymous:
+            type_raw_name = "any /* %s */" % ','.join(parameter_type._values)
+        else:            
+            type_raw_name = stylized_name_for_enum_value(parameter_type.raw_name())
+            if parameter_type.type_domain() != domain:
+                type_raw_name = '{}Domain.{}'.format(parameter_type.type_domain().domain_name, type_raw_name)
+    elif isinstance(parameter_type, PrimitiveType): 
+        raw_name = parameter_type.raw_name()   
+        ts_name = _PRIMITIVE_TO_TS_NAME_MAP.get(raw_name)
+        type_raw_name = ts_name if ts_name else raw_name
+
+    return type_raw_name
+
+class TypeScriptInterfaceGenerator(Generator):
+    def __init__(self, model):
+        Generator.__init__(self, model, "")
+    
+    def output_filename(self):
+        return "InspectorBackendCommands.ts"
+
+    def generate_output(self):
+        sections = []
+        sections.append('''interface FrontendDispatcher { 
+    dispatch(message: string): void; 
+}
+        ''')
+        sections.append('''class Dispatcher {
+    dispatch(method: string, argumentsMessage: string) {
+        var json = JSON.parse(argumentsMessage);
+        var methodArguments = [];        
+        var params = json.params;
+        for (var property in params) {
+            var methodArgument = params[property];
+
+            methodArguments.push(methodArgument);
+        }
+
+        if (this[method] != null) {
+            var result = this[method].apply(this, methodArguments);
+            return JSON.stringify(result);
+        } else {
+            throw new Error("Method " + method + " not found");
+        }
+    }
+}''')
+        sections.extend(map(self.generate_domain, Generator.domains_to_generate(self)))
+        return "\n\n".join(sections)
+
+    def generate_domain(self, domain):
+        lines = []
+        lines.append('// %s.' % domain.domain_name)
+        lines.append('namespace %sDomain {' % domain.domain_name)
```

Comment:
You can also add the description of the domain (if available) like you did for the other models.